edi/models/edi_exchange_consumer_mixin.py

```diff
@@ -19,6 +25,104 @@ class EDIExchangeConsumerMixin(models.AbstractModel):
+            ):
+                continue
+            if exchange_type.enable_snippet:
+                eval_ctx = {
+                    "record": self,
+                    "exchange_type": exchange_type,
+                }
+                eval_ctx.update(self._get_eval_context())
+                safe_eval(
+                    exchange_type.enable_snippet, eval_ctx, mode="exec", nocopy=True
+                )
+                if not eval_ctx.get("result", False):
+                    continue
+            result[exchange_type.id] = exchange_type.display_name
+        return result
+
+    def _get_eval_context(self):
+        """ Prepare the context used when evaluating python code
+            :returns: dict -- evaluation context given to safe_eval
+        """
+        return {
+            "datetime": datetime,
+            "dateutil": dateutil,
+            "time": time,
+            "uid": self.env.uid,
+            "user": self.env.user,
+        }
+
+    @api.model
+    def fields_view_get(
+        self, view_id=None, view_type="form", toolbar=False, submenu=False
+    ):
+        res = super().fields_view_get(
+            view_id=view_id, view_type=view_type, toolbar=toolbar, submenu=submenu
+        )
+        if view_type == "form":
+            doc = etree.XML(res["arch"])
+            for node in doc.xpath("//sheet"):
+                group = False
+                if hasattr(self, "_edi_generate_group"):
+                    group = self._edi_generate_group
+                str_element = self.env["ir.qweb"].render(
+                    "edi.edi_exchange_consumer_mixin_buttons", {"group": group},
+                )
+                node.addprevious(etree.fromstring(str_element))
+            View = self.env["ir.ui.view"]
+
+            # Override context for postprocessing
+            if view_id and res.get("base_model", self._name) != self._name:
+                View = View.with_context(base_model_name=res["base_model"])
+            new_arch, new_fields = View.postprocess_and_fields(self._name, doc, view_id)
+            res["arch"] = new_arch
+            res["fields"] = new_fields
+        return res
+
+    def _generate_edi_configuration_vals(self, exchange_type):
+        return {
+            "model": self._name,
+            "res_id": self.id,
+        }
+
+    def generate_edi_configuration(self, exchange_type_id):
+        self.ensure_one()
+        exchange_type = self.env["edi.exchange.type"].browse(exchange_type_id)
+        backend = exchange_type.backend_id
+        exchange_record = backend.create_record(
+            exchange_type.code, self._generate_edi_configuration_vals(exchange_type)
+        )
+        component = backend._get_component(exchange_record, "manual")
+        # TODO: Define which key the component should have
+        if component:
+            # TODO: Define the function name properly
+            component.manual_generation()
```

Comment:
why another component? What's manual? I think the action should depend on what is missing:
1. missing a record? create it
2. missing data? Receive it
3. missing send? send it
etc...

Reply:
All in all it's not clear what's the purpose of this method :thinking: :sweat_smile:

Reply:
Yes, you are right, I will add an event just in case and keeping it the simplest way :wink: